File: app-geekhome/src/main/java/eu/automateeverything/rest/ClassGsonTypeAdapter.java
package eu.automateeverything.rest;

import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

import java.io.IOException;

public class ClassGsonTypeAdapter extends TypeAdapter<Class> {

    @Override
    public Class read(final JsonReader in) throws IOException {
        String clazz = in.nextString();
        try {
            return Class.forName(clazz);
        } catch (ClassNotFoundException e) {
            throw new IOException(e);
        }
    }

    @Override
    public void write(final JsonWriter out, final Class clazz) throws IOException {
        out.value(clazz.getSimpleName());
    }
}

File: ae-backend/src/main/java/eu/automateeverything/rest/ClassGsonTypeAdapter.kt
package eu.automateeverything.rest

import com.google.gson.TypeAdapter
import com.google.gson.stream.JsonReader
import com.google.gson.stream.JsonWriter
import kotlin.Throws
import java.io.IOException
import java.lang.ClassNotFoundException

class ClassGsonTypeAdapter : TypeAdapter<Class<*>>() {
    @Throws(IOException::class)
    override fun read(`in`: JsonReader): Class<*> {
        val clazz = `in`.nextString()
        return try {
            Class.forName(clazz)
        } catch (e: ClassNotFoundException) {
            throw IOException(e)
        }
    }

    @Throws(IOException::class)
    override fun write(out: JsonWriter, clazz: Class<*>) {
        out.value(clazz.simpleName)
    }
}